### ISSUE TITLE
Python print unexpected output under ranger shell

### ISSUE THREAD

**Issue by tsedii** [ranger/ranger#2135]
My script fill the file called "numbers 1.txt" by sequence from 1 to 10.

Then the script calculate and print out the sum of all numbers from the file.

When I've tried to execute this script:

`

     def FillNumbersFile(numbers_file_name, numbers):
       with open(numbers_file_name, 'w') as numbers_file:
           for index in range(1, numbers + 1):
               numbers_file.write('%d\n' % index)

     def PrintFileNumbersSum(numbers_file_name):
       # define an empty list
       numbers = []

       # open file and read the content in a list
       with open(numbers_file_name, 'r') as numbers_file:
           for line in numbers_file:
               # remove linebreak which is the last character of the string
               current_number = int(line[:-1])

               # add item to the list
               numbers.append(current_number)

        print(sum(numbers))

      numbers_file_name = "numbers1.txt"
      number_count = 10

      FillNumbersFile(numbers_file_name, number_count)
      PrintFileNumbersSum(numbers_file_name)

`
I've got output:

On Ubuntu 20.04.1 LTS under ranger shell with command:
:shell -w python3 myScript
`55nger`

On Ubuntu 20.04.1 LTS under Ubuntu shell:
`55`

On Windows 10:
`55`

#### Runtime Environment

- Operating system and version: 
Ubuntu 20.04.1 LTS
- Terminal emulator and version: 
- Python version: 
3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
- Ranger version/commit: 
ranger 1.9.3
- Locale: 
uk_UA.UTF-8

#### Current Behavior
Print
`55nger`
#### Expected Behavior
Print
`55`
#### Context

I've got unexpected output of the above script.
#### Possible Solutions
???
#### Steps to reproduce
Run above script under the ranger shell by command
:shell -w python3 myScript

**Comment by toonn** [comment]
Minimal reproducer: `shell -w python3 -c 'print(55)'`

**Comment by Nova-Striker** [comment]
@toonn  , I would like to work on this if no one is working on it

**Comment by toonn** [comment]
@Nova-Striker, that would be great! ❤️

**Comment by Nova-Striker** [comment]
@toonn  . Is the bug fixed already ? I ran `ranger` from the source and I was not able to replicate this bug. It only showed the bug when the `ranger` installed in my system was run . I could not find any merged PRs that fixes #2135 . So I thought it was better to ask if the bug is fixed.

**Comment by toonn** [comment]
@Nova-Striker, hmm, I could've sworn I managed to reproduce this not 2 weeks ago. 🤔

**Comment by zaeph** [comment]
You can still reproduce it with double-quotes:
```
:shell -w python3 -c "print(55)"
# Produces "55nger"
```

**Comment by toonn** [comment]
o.O

The script in the OP also still reproduces the issue.
And so does putting `55` in a file and then running `:shell -w cat 55.txt`.

**Comment by toonn** [comment]
Hmm, now if I do double quotes, the first time it doesn't work, the second and third it does and if I then do single quotes it does. Definitely something weird going on.

**Comment by Nova-Striker** [comment]
![Peek 2020-11-07 19-39](https://user-images.githubusercontent.com/64089730/98443405-14bd5a80-2131-11eb-8214-30e1b15f7dd3.gif)


The left is the ranger installed in the system and the right is the ranger from the `ranger.py` file. The right one does not produces the bug even after multiple attempts . Very weird :thinking:  @toonn @zaeph

**Comment by toonn** [comment]
@Nova-Striker, did you try the actual script in the OP or my cat?

**Comment by Nova-Striker** [comment]
I tried the [minimal reproducer](https://github.com/ranger/ranger/issues/2135#issuecomment-716223156)

**Comment by toonn** [comment]
Yeah but the original and the cat reproducer might show different results. All of them are reproducing it for me on master though.

**Comment by Nova-Striker** [comment]
I tried the `cat` one also. The bug does not appears for me in the `master`  

-----
![Peek 2020-11-07 19-57](https://user-images.githubusercontent.com/64089730/98443774-a332db80-2133-11eb-88a8-bf6e28b26493.gif)

**Comment by toonn** [comment]
Hmm, and `55.txt` does contain two characters, right?
I'm running fish and zaeph too maybe this is a fish issue?

**Comment by Nova-Striker** [comment]
Ya , 55.txt contains the exact output of `cat` . Maybe this could be fishy thing

**Comment by toonn** [comment]
Hmm, I can reproduce in both bash and zsh, inside fish though so it could still be environment related I guess.

**Comment by Nova-Striker** [comment]
I would still like to work on it. Can you tell me which part of the code handles the `shell` command ?

**Comment by toonn** [comment]
@Nova-Striker, `ranger/config/commands.py` -> `class shell(Command):`.
